edi/models/edi_exchange_record_mixin.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2020 ACSONE SA
+# Copyright 2020 Creu Blanca
+# @author Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class EDIExchangeRecordMixin(models.AbstractModel):
+    """Record that might have related EDI Exchange records
+    """
+
+    _name = "edi.exchange.record.mixin"
+    _description = "Abstract record where exchange records can be assigned"
+
+    exchange_record_ids = fields.One2many(
+        "edi.exchange.record",
+        inverse_name="res_id",
+        domain=lambda r: [("model", "=", r._name)],
+    )
+    exchange_record_count = fields.Integer(compute="_compute_exchange_record_count")
+    missing_exchange_records = fields.Boolean(
+        compute="_compute_has_missing_exchange_records"
+    )
+
+    def _get_exchange_record_configuration(self):
+        """ This function should return a list of pairs containing a exchange_type_code
+        and a backend. It might contain a dict with the values for creation.
+         To be overridden by the modules"""
+        return []
+
+    def _has_exchange_record_domain(self, exchange_type, backend):
+        return [
+            ("model", "=", self._name),
+            ("res_id", "=", self.id),
+            ("backend_id", "=", backend.id),
+            ("type_id.code", "=", exchange_type),
+        ]
+
+    def _has_exchange_record(self, exchange_type, backend):
+        return self.env["edi.exchange.record"].search(
+            self._has_exchange_record_domain(exchange_type, backend), limit=1
+        )
+
+    @api.depends("exchange_record_ids")
+    def _compute_has_missing_exchange_records(self):
```

Comment:
I don't get how you determine this reliably.

Reply:
Just checking the possible configurations :smile: Now it is clearer